.github/workflows/do-not-merge.yml

```
@@ -0,0 +1,17 @@
+# Fails the pull request if it has the "DO-NOT-MERGE" label
+
+name: Check "DO-NOT-MERGE" label
+
+on:
+  pull_request:
+    types: [opened, reopened, labeled, unlabeled, synchronize]
+
+jobs:
+  do-not-merge:
+    runs-on: ubuntu-latest
+    if: ${{ contains(github.event.*.labels.*.name, 'DO-NOT-MERGE') }}
```

Nit: The only event trigger here is `pull_request`, so the GitHub event here should always be of this type: https://docs.github.com/en/webhooks-and-events/webhooks/webhook-events-and-payloads#pull_request

I think that means we can specify `pull_request` directly here in place of this first asterisk. We only need the second asterisk, to check every label entry in the array.

```suggestion
    if: ${{ contains(github.event.pull_request.labels.*.name, 'DO-NOT-MERGE') }}
```

---

LGTM! check continues to work as expected